tests/core/test_save_files.py

```diff
@@ -64,3 +64,19 @@ def test_save_file():
         in result.stderr.decode()
     )
     assert result.returncode == 1
+
+
+def test_save_cloud_file():
+    # should be no key for cloud paths
+    result = subprocess.run(
+        "lamin save s3://lamindb-ci/lndb-storage/testfile.hdf5 --key filekey.h5ad",
+        shell=True,
+        check=False,
+    )
+    assert result.returncode == 1
```

Comment:
Also didn't notice that there are no tests for logging messages that ensure the artifact was actually created as expected but I guess the return code might be sufficient.

Reply:
Yes, here it should actually fail. And in the next block if it fails, `check=True` will take care of it.